client-api/src/main/java/org/ostelco/topup/api/resources/ProductsResource.java

```diff
@@ -24,50 +24,50 @@
 import lombok.NonNull;
 
 /**
- * Offers API.
+ * Products API.
  *
  */
 @AllArgsConstructor
-@Path("/offers")
-public class OffersResource extends ResourceHelpers {
+@Path("/products")
+public class ProductsResource extends ResourceHelpers {
 
     @NonNull
     private final SubscriberDAO dao;
 
     @GET
     @Produces({"application/json"})
-    public Response getOffers(@Auth AccessTokenPrincipal token) {
+    public Response getProducts(@Auth AccessTokenPrincipal token) {
         if (token == null) {
             return Response.status(Response.Status.UNAUTHORIZED)
                 .build();
         }
 
-        Either<Error, List<Offer>> result = dao.getOffers(token.getName());
+        Either<Error, List<Product>> result = dao.getProducts(token.getName());
 
         return result.isRight()
             ? Response.status(Response.Status.OK)
-                 .entity(getOffersAsJson(result.right().get()))
+                 .entity(getProductsAsJson(result.right().get()))
                  .build()
             : Response.status(Response.Status.NOT_FOUND)
                  .entity(getErrorAsJson(result.left().get()))
                  .build();
     }
 
     @PUT
-    @Path("{offer-id}")
+    @Path("{product-id}")
     @Produces({"application/json"})
-    public Response updateOffer(@Auth AccessTokenPrincipal token,
+    public Response updateProduct(@Auth AccessTokenPrincipal token,
             @NotNull
-            @PathParam("offer-id") String offerId,
+            @PathParam("product-id") String productId,
             @DefaultValue("true") @QueryParam("accepted") boolean accepted) {
```

Comment:
I think we should have `purchaseProduct` as `POST` method and without `boolean accepted`.
Also, actual API might be much more complex than this since it needs integration with Payment Gateway.